### PR TITLE
make the example code more re-usable

### PR DESCRIPTION
**Pull request by sandsmark**
Basically just restructuring the files so that it is easier for people to re-use the Proxy class for their own applications (makes it fairly plug-n-play to enable FakeVim on any QTextEdit/QPlainTextEdit).

### CODE REVIEW

**Comment by hluk** [comment]
:-o Very nice.

File: example/main.cpp [hunk 1]
```diff
@@ -46,7 +46,28 @@ int main(int argc, char *argv[])
     initMainWindow(&mainWindow, editor, usePlainTextEdit ? "QPlainTextEdit" : "QTextEdit");
 
     // Connect slots to FakeVimHandler signals.
-    connectSignals(&handler, &mainWindow, editor, fileToEdit);
+    Proxy *proxy = connectSignals(&handler, &mainWindow, editor);
+
+    QObject::connect(proxy, &Proxy::handleInput,
+        handler, [handler] (const QString &text) { handler->handleInput(text); });
```
Comment (hluk) [inline example/main.cpp:52]: The `handler` is not a pointer and copying QObjects is disabled:

```c++
    QObject::connect(proxy, &Proxy::handleInput,
        &handler, [&handler] (const QString &text) { handler.handleInput(text); });
```

File: example/main.cpp [hunk 2]
```diff
@@ -60,6 +81,13 @@ int main(int argc, char *argv[])
 #endif
     if (QFile::exists(vimrc)) {
         handler.handleCommand("source " + vimrc);
+    } else {
+        // Set some Vim options.
+        handler->handleCommand(_("set expandtab"));
+        handler->handleCommand(_("set shiftwidth=8"));
+        handler->handleCommand(_("set tabstop=16"));
+        handler->handleCommand(_("set autoindent"));
+        handler->handleCommand(_("set smartindent"));
```
Comment (hluk) [inline example/main.cpp:90]: These should be: `handler.handleCommand(QLatin1String(...));`

**Review by hluk — CHANGES_REQUESTED** [review]
Interesting, `./fakevim_example FILE_TO_EDIT` sometimes opens an empty document and not the file. Not sure if it's something that was broken before.

File: example/main.cpp (hunk 1 above)
Comment (sandsmark) [reply example/main.cpp:52]: ... for some reason I didn't enable building examples while testing, so that's embarassing.

**Comment by sandsmark** [comment]
tried squashing a bit, but messed it up, so I reset to the original and just kept the minor commits.

**Comment by hluk** [comment]
Thanks a lot! I'll try to set up GitHub Actions to build and test and let you know.

**Pull request merged**